#!/usr/bin/env python
PACKAGE = "cnbiros_shared_navigation"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Obstacles parameters
gen.add("param_obstacle_strength",	double_t, 0, "Strength of obstacles",		0.2, 0, 10)
gen.add("param_obstacle_decay",		double_t, 0, "Distance decay of obstacles", 0.5, 0, 10)
gen.add("param_obstacle_occupancy", double_t, 0, "Occupancy of obstacles",		9.5, 0, 50)

# User parameters
gen.add("param_command_timeout",	double_t, 0, "Timeout for user commands",		8.0, 0, 20)
gen.add("param_max_goal_distance",	double_t, 0, "Goal distance for user commands", 2.0, 0, 5)
gen.add("param_slope_distance",		double_t, 0, "Decay of user goal",				5.0, 0.1, 10)

# Node parameters
gen.add("param_update_rate",	double_t, 0, "Node update rate", 2.0, 0.1, 20)

exit(gen.generate(PACKAGE, "dynamic_goals_parameters", "DynamicGoals"))
